sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java

```diff
@@ -285,16 +286,23 @@ private static RelDataType toRelDataType(
    * SQL-Java type mapping, with specified Beam rules: <br>
    * 1. redirect {@link AbstractInstant} to {@link Date} so Calcite can recognize it.
```

Comment:
We should probably update this comment to explain how/why ByteString and List are special cases.

Reply:
I updated the List case. The ByteString case wasn't added by me so I don't have context around it.